HelloTorchIOS/HelloTorchIOS/SwitchCounter.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import AVFoundation
+
+class SwitchCounter {
+    
+    private let lightsLogic = LightsLogic()
+    private var counter: Int = 0
+    private var timer: Timer = Timer()
+    private var frequency: Double = 500
+    private var start = DispatchTime.now()
+    private var end = DispatchTime.now()
```

Comment:
`DispatchTime` is a specific domain model for time in the "Grand Central Dispatch" class cluster (GCD is somewhat similar to `Executor` and `ExecutorService` + related classes). It's probably best you don't use it outside GCD APIs

You should probably instead just use `Date()`, which initialises a date set to now. There's also a bunch of helpful formatting tools around such as DateFormatter and DateIntervalFormatter. You can see the Date/Time class cluster here https://developer.apple.com/documentation/foundation/dates_and_times?language=swift

Reply:
Here's some information about GCD https://developer.apple.com/documentation/DISPATCH